muxers/yamux/src/lib.rs

```diff
@@ -42,8 +44,20 @@ pub struct Yamux<S> {
     incoming: S,
     /// Handle to control the connection.
     control: yamux::Control,
+    /// Temporarily buffers inbound streams in case our node is performing backpressure on the remote.
+    ///
+    /// The only way how yamux can make progress is by driving the [`Incoming`] stream. However, the
+    /// [`StreamMuxer`] interface is designed to allow a caller to selectively make progress via
+    /// [`StreamMuxer::poll_inbound`] and [`StreamMuxer::poll_outbound`] whilst the move general
+    /// [`StreamMuxer::poll`] is designed to make progress on existing streams etc.
+    ///
+    /// This buffer stores inbound streams that are created whilst [`StreamMuxer::poll`] is called.
+    /// Once the buffer is full, new inbound streams are dropped.
+    inbound_stream_buffer: VecDeque<yamux::Stream>,
 }
 
+const MAX_BUFFERED_INBOUND_STREAMS: usize = 25;
```

Comment:
If I am not mistaken, the additional buffer of 25 streams increases the overall number of possible inflight inbound streams, correct? In other words, while we previously only supported `max_negotiating_inbound_streams` before dropping streams, we now allow up to `max_negotiating_inbound_streams + 25`.

Is this correct?

Reply:
Yep, that sounds correct to me! Like I said, this number is completely arbitrary and we can change it to something else :)